### PR TITLE
Change how language is added to select list. Instead of using 'contai…

### PR DESCRIPTION
**Pull request by lbandera-ibm-research**
Instead of using 'contains' to check if a list item already exists in the selection list, use a filter to look for exact match so partial matches aren't considered to be already in the list. E.g. Arabic wasn't added to select list since Egyptian Arabic exists in list and the contains selector was a match.

### CODE REVIEW

File: public/demo.js [hunk 1]
```diff
@@ -367,9 +374,9 @@ $(document).ready(function () {
     // Fallback: search for a language-only match (e.g. en to es)
     source = source.split('-').shift(); // split around '-' to handle cases like arz where the language part is actually 3 letters
     target = target.split('-').shift();
-    for (var y in modelList) {
+    for (y in modelList) {
```
Comment (germanattanasio) [inline public/demo.js]: the `var` is important as it will declare the variable in the loop scope

File: public/demo.js [hunk 2]
```diff
@@ -270,14 +275,16 @@ $(document).ready(function () {
 
     // Update output dropdown menu with target language
     $.each(sourceList, function (index, value) {
-      //console.log(value.source + ' source value ' +  $("#dropdownMenuInput").text());
       if (value.source == $.trim($('#dropdownMenuInput').text())) {
-        if (!($('#ulTargetLang li:contains(\'' + value.target + '\')').length)) {
-          //console.log('source lang already exist in li list');
-          //}
-          //else {
+        var search = ($('#ulTargetLang li a').filter(function() {
```
Comment (germanattanasio) [inline public/demo.js]: use `find` instead of `filter`

File: public/demo.js [hunk 3]
```diff
@@ -367,9 +374,9 @@ $(document).ready(function () {
     // Fallback: search for a language-only match (e.g. en to es)
     source = source.split('-').shift(); // split around '-' to handle cases like arz where the language part is actually 3 letters
     target = target.split('-').shift();
-    for (var y in modelList) {
+    for (y in modelList) {
       if (modelList[y].hasOwnProperty('domain')) {
-        var modelListDomain = modelList[y].domain.toString();
+        modelListDomain = modelList[y].domain.toString();
```
Comment (germanattanasio) [inline public/demo.js]: declare the variable with `var`

**Review by germanattanasio — APPROVED** [review]

**Review by germanattanasio — CHANGES_REQUESTED** [review]
please update the code

File: public/demo.js (hunk 1 above)
Comment (nfriedly) [reply public/demo.js]: This was actually a fix for a bug I introduced the other day - I double-defined `y` (and it failed the linter - see #41). The linter would have failed this PR if `y` wasn't declared at all. 

However, the better fix was to rename `y` to a new var name, not re-use the `y` from above. (I would have merged the fix right away, but I was asked to wait.) 

Now that #41 is merged, this change is no longer needed

**Comment by nfriedly** [comment]
I fixed the merge conflict which resolved the `var` confusion.

**Comment by lbandera-ibm-research** [comment]
@germanattanasio @nfriedly - Please review lines 258 -274 in function fillInDropdown. I removed the filter and used find as suggested in the previous review. If you approve, I will make the corresponding change in function updateOutputDropdownMenu.

**Comment by lbandera-ibm-research** [comment]
@nfriedly @germanattanasio - Changes are complete. If you find don't require additional changes, please approve.

**Review by nfriedly — APPROVED** [review]

**Comment by nfriedly** [comment]
This looks good, sorry for the delays.

**Pull request merged**